roachtest: activerecord failed

**Note:** This build has runtime assertions enabled. If the same failure was hit in a run without assertions enabled, there should be a similar failure without this message. If there isn&#39;t one, then this failure is likely due to an assertion violation or (assertion) timeout.

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/17990793?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/17990793?buildTab=artifacts#/activerecord) on release-24.2 @ [8a1ed05f5ef63eab776779a8625b328e06cc6512](https://github.com/cockroachdb/cockroach/commits/8a1ed05f5ef63eab776779a8625b328e06cc6512):


```
(orm_helpers.go:224).summarizeFailed: 
Tests run on Cockroach v24.2.6-dev-8a1ed05f5ef63eab776779a8625b328e06cc6512
Tests run against activerecord 7.2.1
8487 Total Tests Run
8463 tests passed
24 tests failed
60 tests skipped
5 tests ignored
0 tests passed unexpectedly
1 test failed unexpectedly
0 tests expected failed but skipped
0 tests expected failed but not run
---
--- PASS: SerializedAttributeTestWithYamlSafeLoad#test_unexpected_serialized_type (expected)
--- PASS: SerializedAttributeTest#test_unexpected_serialized_type (expected)
--- FAIL: CockroachDB::FixturesTest#test_create_fixtures (unexpected)
For a full summary look at the activerecord artifacts 
An updated blocklist (activeRecordBlocklist) is available in the artifacts' activerecord log
test artifacts and logs in: /artifacts/activerecord/run_1
```

Parameters:
 - <code>arch=amd64</code>
 - <code>cloud=gce</code>
 - <code>coverageBuild=false</code>
 - <code>cpu=4</code>
 - <code>encrypted=false</code>
 - <code>fs=ext4</code>
 - <code>localSSD=true</code>
 - <code>runtimeAssertionsBuild=true</code>
 - <code>ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-17990793/activerecord/1733122259794/1733130295810)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #136066 roachtest: activerecord failed [PostGISTest#test_point_to_json] [C-test-failure O-roachtest O-robot P-2 T-sql-foundations branch-master]
</p>
</details>
/cc @cockroachdb/sql-foundations
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-45079

roachtest.activerecord [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/18041202?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/18041202?buildTab=artifacts#/activerecord) on release-24.2 @ [8148399ab37d1c82ac2b46067fa19848da00d667](https://github.com/cockroachdb/cockroach/commits/8148399ab37d1c82ac2b46067fa19848da00d667):


```
0 tests expected failed but skipped
0 tests expected failed but not run
---
--- FAIL: TokenForTest#test_finds_record_by_token (unexpected)
--- FAIL: TokenForTest#test_subclasses_can_redefine_tokens (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_accepts_any_object_that_implements_to_h (unexpected)
--- FAIL: TokenForTest#test_raises_when_no_primary_key_has_been_declared (unexpected)
--- FAIL: TokenForTest#test_finds_record_with_a_custom_primary_key (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_short_circuits_when_password_is_nil (unexpected)
--- FAIL: TokenForTest#test_does_not_find_record_when_token_has_expired (unexpected)
--- FAIL: TokenForTest#test_raises_when_token_definition_does_not_exist (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_does_not_authenticate_when_password_is_incorrect (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_requires_at_least_one_attribute (unexpected)
--- FAIL: SuppressorTest#test_suppresses_validations_on_create (unexpected)
--- FAIL: TokenForTest#test_finds_record_through_relation (unexpected)
--- FAIL: TokenForTest#test_does_not_find_record_when_expires_in_is_different (unexpected)
--- FAIL: HasManyThroughAssociationsTest#test_has_many_through_with_scope_should_respect_table_alias (unexpected)
--- FAIL: SecureTokenTest#test_token_on_callback (unexpected)
--- FAIL: SecureTokenTest#test_token_values_are_generated_for_specified_attributes_and_persisted_on_save (unexpected)
--- FAIL: SecureTokenTest#test_generating_token_on_initialize_happens_only_once (unexpected)
--- FAIL: SecureTokenTest#test_regenerating_the_secure_token (unexpected)
--- FAIL: SuppressorTest#test_suppresses_create_in_callback (unexpected)
--- FAIL: SecureTokenTest#test_token_value_not_overwritten_when_present (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_requires_at_least_one_password (unexpected)
--- FAIL: SuppressorTest#test_resumes_saving_after_suppression_complete (unexpected)
--- FAIL: SuppressorTest#test_suppresses_update (unexpected)
--- FAIL: TokenForTest#test_does_not_find_record_when_token_is_invalid (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_authenticates_when_password_is_correct (unexpected)
--- FAIL: TokenForTest#test_supports_JSON-serializable_embedded_data (unexpected)
--- FAIL: TokenForTest#test_returns_nil_when_record_is_not_found (unexpected)
--- FAIL: TokenForTest#test_does_not_find_record_when_embedded_data_is_different (unexpected)
--- PASS: SerializedAttributeTest#test_unexpected_serialized_type (expected)
--- FAIL: TokenForTest#test_raises_on_bang_when_record_is_not_found (unexpected)
--- FAIL: TokenForTest#test_finds_record_through_subclass (unexpected)
--- FAIL: HasAndBelongsToManyAssociationsTest#test_association_name_is_the_same_as_join_table_name (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_takes_the_same_amount_of_time_regardless_of_whether_record_is_found (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_authenticates_using_multiple_passwords (unexpected)
--- FAIL: TokenForTest#test_tokens_do_not_expire_by_default (unexpected)
--- PASS: SerializedAttributeTestWithYamlSafeLoad#test_unexpected_serialized_type (expected)
--- FAIL: TokenForTest#test_does_not_find_record_when_token_is_for_a_different_purpose (unexpected)
--- FAIL: SecureTokenTest#test_generating_token_on_initialize_is_skipped_if_column_was_not_selected (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_short_circuits_when_password_is_an_empty_string (unexpected)
--- FAIL: TokenForTest#test_finds_record_when_token_has_not_expired_and_embedded_data_has_not_changed (unexpected)
--- FAIL: HasOneAssociationsTest#test_has_one_double_belongs_to_destroys_both_from_either_end (unexpected)
--- FAIL: SecurePasswordTest#test_authenticate_by_finds_record_using_multiple_attributes (unexpected)
--- FAIL: SecureTokenTest#test_token_calls_the_setter_method (unexpected)
--- FAIL: AutomaticInverseFindingTests#test_has_one_and_belongs_to_with_non_default_foreign_key_should_not_find_inverse_automatically (unexpected)
For a full summary look at the activerecord artifacts 
An updated blocklist (activeRecordBlocklist) is available in the artifacts' activerecord log
test artifacts and logs in: /artifacts/activerecord/run_1
```

Parameters:
 - <code>arch=amd64</code>
 - <code>cloud=gce</code>
 - <code>coverageBuild=false</code>
 - <code>cpu=4</code>
 - <code>encrypted=false</code>
 - <code>fs=ext4</code>
 - <code>localSSD=true</code>
 - <code>runtimeAssertionsBuild=false</code>
 - <code>ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-18041202/activerecord/1733381452534/1733389366534)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #136675 roachtest: activerecord failed [C-test-failure O-roachtest O-robot T-sql-foundations branch-release-24.2.6-rc]
- #136562 roachtest: activerecord failed [B-runtime-assertions-enabled C-test-failure O-roachtest O-robot P-2 T-sql-foundations branch-release-24.3 release-blocker]
- #136561 roachtest: activerecord failed [B-runtime-assertions-enabled C-test-failure O-roachtest O-robot P-2 T-sql-foundations branch-release-24.3.1-rc]
- #136560 roachtest: activerecord failed [B-runtime-assertions-enabled C-test-failure O-roachtest O-robot P-2 T-sql-foundations branch-master]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*activerecord.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

--- a/pkg/cmd/roachtest/tests/activerecord_blocklist.go
+++ b/pkg/cmd/roachtest/tests/activerecord_blocklist.go
@@ -24,6 +24,7 @@ var activeRecordBlocklist = blocklist{
 var activeRecordIgnoreList = blocklist{
 	`ActiveRecord::ConnectionAdapters::PostgreSQLAdapterTest#test_translate_no_connection_exception_to_not_established`:                                        "pg_terminate_backend not implemented",
 	`BasicsTest#test_default_values_are_deeply_dupped`:                                                                                                         "flaky",
+	`CockroachDB::FixturesTest#test_create_fixtures`:                                                                                                           "flaky",
 	`TestAutosaveAssociationOnAHasAndBelongsToManyAssociation#test_should_not_save_and_return_false_if_a_callback_cancelled_saving_in_either_create_or_update`: "flaky",
 	`TestAutosaveAssociationOnAHasAndBelongsToManyAssociation#test_should_not_update_children_when_parent_creation_with_no_reason`:                             "flaky",
 	`TestAutosaveAssociationOnAHasAndBelongsToManyAssociation#test_should_update_children_when_autosave_is_true_and_parent_is_new_but_child_is_not`:            "flaky",

